CMakeLists.txt

```diff
@@ -72,7 +72,9 @@ endif()
 find_package(Doxygen)
 if(Doxygen_FOUND)
   set(DOXYGEN_GENERATE_XML YES) 
-  doxygen_add_docs(doxygen core)
+  set(DOXYGEN_WARN_LOGFILE "warnings.log") 
+  set(DOXYGEN_WARN_IF_UNDOCUMENTED NO)
```

Comment:
Actually I think the right thing to do would be to get rid of these and configure with `-DDOXYGEN_GENERATE_XML=YES` when we want to upload the coverage reports and with `-DDOXYGEN_WARN_AS_ERROR=YES -DDOXYGEN_WARN_IF_UNDOCUMENTED=YES` when we want to check they are no warnings.

Reply:
Huh, `DOXYGEN_WARN_IF_UNDOCUMENTED=NO` ? Don't we want warnings for all undocumented members?

Reply:
I'd say no.  I am definitely not going to provide doc for the one that have none.

Reply:
Ok, for me, @sslattery any objections?

Reply:
No objections. I have tried to use doxygen for all relevant public interfaces.

Reply:
@junghans Not sure if I made myself clear.  I am suggesting we drop both lines https://github.com/ECP-copa/Cabana/blob/6f4e48b61ca6640a005b32411011c054c0aca419/CMakeLists.txt#L74-L75 and pass the variables as needed at configuration time.

Reply:
Making `DOXYGEN_WARN_IF_UNDOCUMENTED=NO` the default seems fine to me.

And `DOXYGEN_GENERATE_XML=YES` costs nothing, but space.